---
title: "rnaseqGene"
author: "Nathan Haywood"
date: "3/4/2020"
output: html_document
---

## Loads the airway package
```{r}
## ----loadairway------------------------------------------------------------
library("airway")
```

## Sets the variable "dir" to the pathway where the airway package is locally stored
```{r}
## ----dir-------------------------------------------------------------------
dir <- system.file("extdata", package="airway", mustWork=TRUE)
```

## Lists all of the files and directories in the dir and then specifies for just the quant directories
```{r}
## ----list.files------------------------------------------------------------
list.files(dir)
list.files(file.path(dir, "quants"))
```

## A comma separated values table is assigned to the variable "csvfile" and the first row of the file is extracted and shown.
```{r}
## ----sampleinfo------------------------------------------------------------
csvfile <- file.path(dir, "sample_table.csv")
coldata <- read.csv(csvfile, row.names=1, stringsAsFactors=FALSE)
coldata
```

## A coldata table is created with two columns, names and files, and is populated with the paths to the quant files from the previous section.
```{r}
## ----makecoldata-----------------------------------------------------------
coldata <- coldata[1:2,]
coldata$names <- coldata$Run
coldata$files <- file.path(dir, "quants", coldata$names, "quant.sf.gz")
file.exists(coldata$files)
```

## The tximeta package is loaded and run on the coldata table
```{r}
## ----tximeta, message=TRUE-------------------------------------------------
library("tximeta")
se <- tximeta(coldata)
```

## "se" is a transcript object that was loaded in the tximeta package. The "dim" call checks the dimensions and the "head" call displays the first lines of the object.
```{r}
## ----lookse----------------------------------------------------------------
dim(se)
head(rownames(se))
```

## The summarizeToGene call replicates then transforms the transcript object "se" to "gse"
```{r}
## ----summarize, message=TRUE-----------------------------------------------
gse <- summarizeToGene(se)
```

## The "dim" call checks the dimensions and the "head" call displays the first lines of the gene object.
```{r}
## ----lookgse---------------------------------------------------------------
dim(gse)
head(rownames(gse))
```

## A graph is created with a plotting method to visualize the intersection of the colData object and the range of rows.
```{r}
## ----sumexp, echo=FALSE----------------------------------------------------
par(mar=c(0,0,0,0))
plot(1,1,xlim=c(0,100),ylim=c(0,100),bty="n",
     type="n",xlab="",ylab="",xaxt="n",yaxt="n")
polygon(c(45,90,90,45),c(5,5,70,70),col="pink",border=NA)
polygon(c(45,90,90,45),c(68,68,70,70),col="pink3",border=NA)
text(67.5,40,"assay(s)")
text(67.5,35,'e.g. "counts", ...')
polygon(c(10,40,40,10),c(5,5,70,70),col="skyblue",border=NA)
polygon(c(10,40,40,10),c(68,68,70,70),col="skyblue3",border=NA)
text(25,40,"rowRanges")
polygon(c(45,90,90,45),c(75,75,95,95),col="palegreen",border=NA)
polygon(c(45,47,47,45),c(75,75,95,95),col="palegreen3",border=NA)
text(67.5,85,"colData")
```

## The data call loads the full gse object.
```{r}
## ----loadfullgse-----------------------------------------------------------
data(gse)
gse
```

## The "assayNames" call displays the titles of each column, while "head" shows the first few lines of the object and the "colSums" shows the sum of each column in the object.
```{r}
## ----assaysgse-------------------------------------------------------------
assayNames(gse)
head(assay(gse), 3)
colSums(assay(gse))
```

## Shows the ranges for the first and final five genes.
```{r}
## ----rowrangesgse----------------------------------------------------------
rowRanges(gse)
```

## Shows the metadata related to each of the sequences.
```{r}
## ----lookseqinfo-----------------------------------------------------------
seqinfo(rowRanges(gse))
```

## Shows the information about the "gse" dataframe.
```{r}
## ----coldatagse------------------------------------------------------------
colData(gse)
```

## The "gse$donor" and "gse$condition" displays each of the parameters in the object.
```{r}
## ----gsevars---------------------------------------------------------------
gse$donor
gse$condition
```

## Each of these renames these parameters.
```{r}
## ----gsevarsrename---------------------------------------------------------
gse$cell <- gse$donor
gse$dex <- gse$condition
```

## The "levels(gse$dex) <- c("untrt", "trt")" call casts the names of the parameters to the new vector.
```{r}
## ----renamelevels----------------------------------------------------------
levels(gse$dex)
# when renaming levels, the order must be preserved!
levels(gse$dex) <- c("untrt", "trt")
```

## This loads the "magrittr" package and executes a pipe with the relevel function.
```{r}
## ----gsedex----------------------------------------------------------------
library("magrittr")
gse$dex %<>% relevel("untrt")
gse$dex
```

## This line is  not executed, but is the equilvalent to the previous line of code.
```{r}
## ----explaincmpass, eval = FALSE-------------------------------------------
#  gse$dex <- relevel(gse$dex, "untrt")
```

## The "round" call displays the fragments that can be mapped to the genes.
```{r}
## ----countreads------------------------------------------------------------
round( colSums(assay(gse)) / 1e6, 1 )
```

## The "DESeq2 package is loaded.
```{r}
## ----loaddeseq2------------------------------------------------------------
library("DESeq2")
```

## This call creates a fully annotated dds object for the gene analysis.
```{r}
## ----makedds---------------------------------------------------------------
dds <- DESeqDataSet(gse, design = ~ cell + dex)
```

## This creates a countdata object that creates an assay with the counts from the gse object. The "head" call displays the first lines of the countdata object.
```{r}
## --------------------------------------------------------------------------
countdata <- round(assays(gse)[["counts"]])
head(countdata, 3)
```

## The coldata object is recast to hold the data from the gse object.
```{r}
## --------------------------------------------------------------------------
coldata <- colData(gse)
```

## An object, "ddsMat", is created to hold the colData and countData objects.
```{r}
## --------------------------------------------------------------------------
ddsMat <- DESeqDataSetFromMatrix(countData = countdata,
                                 colData = coldata,
                                 design = ~ cell + dex)
```

## This block is used to filter out the rows that contain one or fewer counts to isolate the data rows that have useful data.
```{r}
## --------------------------------------------------------------------------
nrow(dds)
keep <- rowSums(counts(dds)) > 1
dds <- dds[keep,]
nrow(dds)
```

## This goes to continue filtering the dataset to isolate rows with a higher data density.
```{r}
## --------------------------------------------------------------------------
# at least 3 samples with a count of 10 or higher
keep <- rowSums(counts(dds) >= 10) >= 3
```

## The output of this graph is used to show how the variance in the data grwos as the average does.
```{r}
## ----meanSdCts-------------------------------------------------------------
lambda <- 10^seq(from = -1, to = 2, length = 1000)
cts <- matrix(rpois(1000*100, lambda), ncol = 100)
library("vsn")
meanSdPlot(cts, ranks = FALSE)
```

## This graph is used to show how the log transformed graph has the highest varaince at a significantly lower average.
```{r}
## ----meanSdLogCts----------------------------------------------------------
log.cts.one <- log2(cts + 1)
meanSdPlot(log.cts.one, ranks = FALSE)
```

## This shows the raw values used to create the first graph.
```{r}
## ----vst-------------------------------------------------------------------
vsd <- vst(dds, blind = FALSE)
head(assay(vsd), 3)
colData(vsd)
```

## This shows the raw values used to create the second graph.
```{r}
## ----rlog------------------------------------------------------------------
rld <- rlog(dds, blind = FALSE)
head(assay(rld), 3)
```

## The packages "dplyr" and "ggplot2" are loaded and the samples are plotted against eachother to show the differences.
```{r}
## ----transformplot, fig.width = 6, fig.height = 2.5------------------------
library("dplyr")
library("ggplot2")

dds <- estimateSizeFactors(dds)

df <- bind_rows(
  as_data_frame(log2(counts(dds, normalized=TRUE)[, 1:2]+1)) %>%
         mutate(transformation = "log2(x + 1)"),
  as_data_frame(assay(vsd)[, 1:2]) %>% mutate(transformation = "vst"),
  as_data_frame(assay(rld)[, 1:2]) %>% mutate(transformation = "rlog"))
  
colnames(df)[1:2] <- c("x", "y")  

ggplot(df, aes(x = x, y = y)) + geom_hex(bins = 80) +
  coord_fixed() + facet_grid( . ~ transformation) 
```

## This is used to determine the overall similarity between the samples.
```{r}
## --------------------------------------------------------------------------
sampleDists <- dist(t(assay(vsd)))
sampleDists
```

## The packages "pheatmap" and "RColorBrewer" are loaded.
```{r}
## --------------------------------------------------------------------------
library("pheatmap")
library("RColorBrewer")
```

## This creates a heatmap to show the sample distance matrix.
```{r}
## ----distheatmap, fig.width = 6.1, fig.height = 4.5------------------------
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste( vsd$dex, vsd$cell, sep = " - " )
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors)
```

## The "PoiClaClu" package is loaded and used to do Poisson Distance analysis.
```{r}
## --------------------------------------------------------------------------
library("PoiClaClu")
poisd <- PoissonDistance(t(counts(dds)))
```

## The Poisson Distance analysis is used to create a heatmap for the sample distances. The Poisson distance matrix uses the raw data values instead of the normalized ones.
```{r}
## ----poisdistheatmap, fig.width = 6.1, fig.height = 4.5--------------------
samplePoisDistMatrix <- as.matrix( poisd$dd )
rownames(samplePoisDistMatrix) <- paste( dds$dex, dds$cell, sep=" - " )
colnames(samplePoisDistMatrix) <- NULL
pheatmap(samplePoisDistMatrix,
         clustering_distance_rows = poisd$dd,
         clustering_distance_cols = poisd$dd,
         col = colors)
```

## A Principle Components Analysis plot is created with the to show variance.
```{r}
## ----plotpca, fig.width=6, fig.height=4.5----------------------------------
plotPCA(vsd, intgroup = c("dex", "cell"))
```

## This uses the ggplot2 package to begin working on the PCA plot from the vsd data.
```{r}
## --------------------------------------------------------------------------
pcaData <- plotPCA(vsd, intgroup = c( "dex", "cell"), returnData = TRUE)
pcaData
percentVar <- round(100 * attr(pcaData, "percentVar"))

```

## ggplot2 is used to create the PCA plot with significantly more control voer the details.
```{r}
## ----ggplotpca, fig.width=6, fig.height=4.5--------------------------------
ggplot(pcaData, aes(x = PC1, y = PC2, color = dex, shape = cell)) +
  geom_point(size =3) +
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  coord_fixed() +
  ggtitle("PCA with VST data")
```

## The "glmpca" package is loaded and  creates a generalized PCA plot.
```{r}
## --------------------------------------------------------------------------
library("glmpca")
gpca <- glmpca(counts(dds), L=2)
gpca.dat <- gpca$factors
gpca.dat$dex <- dds$dex
gpca.dat$cell <- dds$cell

## ----glmpca, fig.width=6, fig.height=4.5-----------------------------------
ggplot(gpca.dat, aes(x = dim1, y = dim2, color = dex, shape = cell)) +
  geom_point(size =3) + coord_fixed() + ggtitle("glmpca - Generalized PCA")
```

## An MDS (Multidimensional Scaling) plot is best used with the matrixes of distances instead of the matrixes of data.
```{r}
## ----mdsvst, fig.width=6, fig.height=4.5-----------------------------------
mds <- as.data.frame(colData(vsd))  %>%
         cbind(cmdscale(sampleDistMatrix))
ggplot(mds, aes(x = `1`, y = `2`, color = dex, shape = cell)) +
  geom_point(size = 3) + coord_fixed() + ggtitle("MDS with VST data")
```

## A similar MDS plot can be created with the Poisson Distances.
```{r}
## ----mdspois, fig.width=6, fig.height=4.5----------------------------------
mdsPois <- as.data.frame(colData(dds)) %>%
   cbind(cmdscale(samplePoisDistMatrix))
ggplot(mdsPois, aes(x = `1`, y = `2`, color = dex, shape = cell)) +
  geom_point(size = 3) + coord_fixed() + ggtitle("MDS with PoissonDistances")
```

## This runs a differential expression pipeline on the raw counts.
```{r}
## ----airwayDE--------------------------------------------------------------
dds <- DESeq(dds)
```

## The results of the differential expression pipeline are shown here.
```{r}
## --------------------------------------------------------------------------
res <- results(dds)
res
```

## The more controlled results could also have been achieved by directly specifying the results parameters.
```{r}
## --------------------------------------------------------------------------
res <- results(dds, contrast=c("dex","trt","untrt"))
```

## The "res" dataframe object metadata is shown here.
```{r}
## --------------------------------------------------------------------------
mcols(res, use.names = TRUE)
```

## A summary of the "res" object is shown here.
```{r}
## --------------------------------------------------------------------------
summary(res)
```

## The false discovery rate threshold is lowered and the results function is updated to optimize filtration.
```{r}
## --------------------------------------------------------------------------
res.05 <- results(dds, alpha = 0.05)
table(res.05$padj < 0.05)
```

## This selects for genes that have significant changes by more than a factor of 2 (either 0.5 or 2).
```{r}
## --------------------------------------------------------------------------
resLFC1 <- results(dds, lfcThreshold=1)
table(resLFC1$padj < 0.1)
```

## Here the results of the comparison between two levels are shown.
```{r}
## --------------------------------------------------------------------------
results(dds, contrast = c("cell", "N061011", "N61311"))
```

## Multiple Testing methods are required to provide a P value that disproves the null.
```{r}
## ----sumres----------------------------------------------------------------
sum(res$pvalue < 0.05, na.rm=TRUE)
sum(!is.na(res$pvalue))
```

## Here we are looking for the number of genes with a less than 10% false positive rate.
```{r}
## --------------------------------------------------------------------------
sum(res$padj < 0.1, na.rm=TRUE)
```

## The genes are then arranged by strongest downregulation.
```{r}
## --------------------------------------------------------------------------
resSig <- subset(res, padj < 0.1)
head(resSig[ order(resSig$log2FoldChange), ])
```

## They can also be arranged by the strongest upregulation.
```{r}
## --------------------------------------------------------------------------
head(resSig[ order(resSig$log2FoldChange, decreasing = TRUE), ])
```

## The plotCounts method shows counts for the groupings of the genes.
```{r}
## ----plotcounts------------------------------------------------------------
topGene <- rownames(res)[which.min(res$padj)]
plotCounts(dds, gene = topGene, intgroup=c("dex"))
```

## The "ggbeeswarm" package is loaded and used along with ggplot2 to create a custom plotCount plot.
```{r}
## ----ggplotcountsjitter, fig.width = 4, fig.height = 3---------------------
library("ggbeeswarm")
geneCounts <- plotCounts(dds, gene = topGene, intgroup = c("dex","cell"),
                         returnData = TRUE)
ggplot(geneCounts, aes(x = dex, y = count, color = cell)) +
  scale_y_log10() +  geom_beeswarm(cex = 3)
```

## The normalized counts have connected lines to indicate cell lines impact.
```{r}
## ----ggplotcountsgroup, fig.width = 4, fig.height = 3----------------------
ggplot(geneCounts, aes(x = dex, y = count, color = cell, group = cell)) +
  scale_y_log10() + geom_point(size = 3) + geom_line()
```

## The "apeglm" packge is loaded and a Minus Average plot is created.
```{r}
## ----plotma----------------------------------------------------------------
library("apeglm")
resultsNames(dds)
res <- lfcShrink(dds, coef="dex_trt_vs_untrt", type="apeglm")
plotMA(res, ylim = c(-5, 5))
```

## This is the plot that lacks the log fold change reduction.
```{r}
## ----plotmaNoShr-----------------------------------------------------------
res.noshr <- results(dds, name="dex_trt_vs_untrt")
plotMA(res.noshr, ylim = c(-5, 5))
```

## This shows how to select specific rows or regions of the graph.
```{r}
## ----plotmalabel-----------------------------------------------------------
plotMA(res, ylim = c(-5,5))
topGene <- rownames(res)[which.min(res$padj)]
with(res[topGene, ], {
  points(baseMean, log2FoldChange, col="dodgerblue", cex=2, lwd=2)
  text(baseMean, log2FoldChange, topGene, pos=2, col="dodgerblue")
})
```

## The histogram shows the p value frequencies, and is most effective when smallest values are excluded.
```{r}
## ----histpvalue2-----------------------------------------------------------
hist(res$pvalue[res$baseMean > 1], breaks = 0:20/20,
     col = "grey50", border = "white")
```

## The "genefilter" package is loaded and used to show gene clustering.
```{r}
## --------------------------------------------------------------------------
library("genefilter")
topVarGenes <- head(order(rowVars(assay(vsd)), decreasing = TRUE), 20)
```

## This heatmap shows deviation from the ave as opposed to expression strenght.
```{r}
## ----genescluster----------------------------------------------------------
mat  <- assay(vsd)[ topVarGenes, ]
mat  <- mat - rowMeans(mat)
anno <- as.data.frame(colData(vsd)[, c("cell","dex")])
pheatmap(mat, annotation_col = anno)
```

## New bins are created with the "quantile" function and the p values for each are calculated.
```{r}
## ----sensitivityovermean, fig.width=6--------------------------------------
qs <- c(0, quantile(resLFC1$baseMean[resLFC1$baseMean > 0], 0:6/6))
bins <- cut(resLFC1$baseMean, qs)
levels(bins) <- paste0("~", round(signif((qs[-1] + qs[-length(qs)])/2, 2)))
fractionSig <- tapply(resLFC1$pvalue, bins, function(p)
                          mean(p < .05, na.rm = TRUE))
barplot(fractionSig, xlab = "mean normalized count",
                     ylab = "fraction of small p values")
```

## ## This line is  not executed, but is similar to the previous block of code.
```{r}
## ---- eval=FALSE-----------------------------------------------------------
#  library("IHW")
#  res.ihw <- results(dds, filterFun=ihw)
```


## The "AnnotationDbi" and "org.Hs.eg.db" packages are loaded.
```{r}
## --------------------------------------------------------------------------
library("AnnotationDbi")
library("org.Hs.eg.db")
```

## This is a list of the available key types from the AnnotationDbi database. 
```{r}
## --------------------------------------------------------------------------
columns(org.Hs.eg.db)
```

## This is used to specify what information is desired from the database.
```{r}
## --------------------------------------------------------------------------
ens.str <- substr(rownames(res), 1, 15)
res$symbol <- mapIds(org.Hs.eg.db,
                     keys=ens.str,
                     column="SYMBOL",
                     keytype="ENSEMBL",
                     multiVals="first")
res$entrez <- mapIds(org.Hs.eg.db,
                     keys=ens.str,
                     column="ENTREZID",
                     keytype="ENSEMBL",
                     multiVals="first")
```

## These are the external gene IDs.
```{r}
## --------------------------------------------------------------------------
resOrdered <- res[order(res$pvalue),]
head(resOrdered)
```

## This code is not evaluated.
```{r}
## ----eval=FALSE------------------------------------------------------------
#  resOrderedDF <- as.data.frame(resOrdered)[1:100, ]
#  write.csv(resOrderedDF, file = "results.csv")

## ----eval=FALSE------------------------------------------------------------
#  library("ReportingTools")
#  htmlRep <- HTMLReport(shortName="report", title="My report",
#                        reportDirectory="./report")
#  publish(resOrderedDF, htmlRep)
#  url <- finish(htmlRep)
#  browseURL(url)
```

## This plots the differential expression results in the genomic space.
```{r}
## --------------------------------------------------------------------------
resGR <- lfcShrink(dds, coef="dex_trt_vs_untrt", type="apeglm", format="GRanges")
resGR
```

## Symbols are added to include gene labels on the plot
```{r}
## --------------------------------------------------------------------------
ens.str <- substr(names(resGR), 1, 15)
resGR$symbol <- mapIds(org.Hs.eg.db, ens.str, "SYMBOL", "ENSEMBL")
```

## The "Gviz" package is loaded.
```{r}
## --------------------------------------------------------------------------
library("Gviz")
```

## A million base pairs window on either side of the gene with the smallest p value is set to collect a subset of genes within that window. 
```{r}
## --------------------------------------------------------------------------
window <- resGR[topGene] + 1e6
strand(window) <- "*"
resGRsub <- resGR[resGR %over% window]
naOrDup <- is.na(resGRsub$symbol) | duplicated(resGRsub$symbol)
resGRsub$group <- ifelse(naOrDup, names(resGRsub), resGRsub$symbol)
```

## A vector identifying if the "padj" value of the gene was low.
```{r}
## --------------------------------------------------------------------------
status <- factor(ifelse(resGRsub$padj < 0.05 & !is.na(resGRsub$padj),
                        "sig", "notsig"))
```

## A plot showing the locations of each gene on is created with functions from the "Gviz" package.
```{r}
## ----gvizplot--------------------------------------------------------------
options(ucscChromosomeNames = FALSE)
g <- GenomeAxisTrack()
a <- AnnotationTrack(resGRsub, name = "gene ranges", feature = status)
d <- DataTrack(resGRsub, data = "log2FoldChange", baseline = 0,
               type = "h", name = "log2 fold change", strand = "+")
plotTracks(list(g, d, a), groupAnnotation = "group",
           notsig = "grey", sig = "hotpink")
```

## The "sva" package is loaded.
```{r}
## --------------------------------------------------------------------------
library("sva")
```

## A matrix with the normalized counts is created for samples where the average count was greater than 1.
```{r}
## --------------------------------------------------------------------------
dat  <- counts(dds, normalized = TRUE)
idx  <- rowMeans(dat) > 1
dat  <- dat[idx, ]
mod  <- model.matrix(~ dex, colData(dds))
mod0 <- model.matrix(~   1, colData(dds))
svseq <- svaseq(dat, mod, mod0, n.sv = 2)
svseq$sv
```

## This is used to check how well the "SVA" method recovered the sources of variation.
```{r}
## ----svaplot---------------------------------------------------------------
par(mfrow = c(2, 1), mar = c(3,5,3,1))
for (i in 1:2) {
  stripchart(svseq$sv[, i] ~ dds$cell, vertical = TRUE, main = paste0("SV", i))
  abline(h = 0)
 }
```

## This is used to prevent the "SVA" method from impacting the counts.
```{r}
## --------------------------------------------------------------------------
ddssva <- dds
ddssva$SV1 <- svseq$sv[,1]
ddssva$SV2 <- svseq$sv[,2]
design(ddssva) <- ~ SV1 + SV2 + dex
```

## The "RUVSeq" package is loaded.
```{r}
## --------------------------------------------------------------------------
library("RUVSeq")
```

## Usng the "RUVg" method, it's possible to estimate the factors of unwanted variation, which is similar to the "SVA" surrogate variables.
```{r}
## --------------------------------------------------------------------------
set <- newSeqExpressionSet(counts(dds))
idx  <- rowSums(counts(set) > 5) >= 2
set  <- set[idx, ]
set <- betweenLaneNormalization(set, which="upper")
not.sig <- rownames(res)[which(res$pvalue > .1)]
empirical <- rownames(set)[ rownames(set) %in% not.sig ]
set <- RUVg(set, empirical, k=2)
pData(set)
```

## This shows how the factors can be estimated by "RUV".
```{r}
## ----ruvplot---------------------------------------------------------------
par(mfrow = c(2, 1), mar = c(3,5,3,1))
for (i in 1:2) {
  stripchart(pData(set)[, i] ~ dds$cell, vertical = TRUE, main = paste0("W", i))
  abline(h = 0)
 }
```

## It is possible to control these factors by defining them.
```{r}
## --------------------------------------------------------------------------
ddsruv <- dds
ddsruv$W1 <- set$W_1
ddsruv$W2 <- set$W_2
design(ddsruv) <- ~ W1 + W2 + dex
```

## The "fission" package is loaded, and a basic time course analysis is run. 
```{r}
## --------------------------------------------------------------------------
library("fission")
data("fission")
ddsTC <- DESeqDataSet(fission, ~ strain + minute + strain:minute)
```

## This is a likelihood ratio test where the strain specific differences are removed over time.
```{r}
## ----fissionDE-------------------------------------------------------------
ddsTC <- DESeq(ddsTC, test="LRT", reduced = ~ strain + minute)
resTC <- results(ddsTC)
resTC$symbol <- mcols(ddsTC)$symbol
head(resTC[order(resTC$padj),], 4)
```

## ggplot2 is used to show the counts for each gorup over a time frame.
```{r}
## ----fissioncounts, fig.width=6, fig.height=4.5----------------------------
fiss <- plotCounts(ddsTC, which.min(resTC$padj), 
                   intgroup = c("minute","strain"), returnData = TRUE)
fiss$minute <- as.numeric(as.character(fiss$minute))
ggplot(fiss,
  aes(x = minute, y = count, color = strain, group = strain)) + 
  geom_point() + stat_summary(fun.y=mean, geom="line") +
  scale_y_log10()
```

## A "Wald" test for the log2 fold changes at individual time points.
```{r}
## --------------------------------------------------------------------------
resultsNames(ddsTC)
res30 <- results(ddsTC, name="strainmut.minute30", test="Wald")
res30[which.min(resTC$padj),]
```

## Cluster significant genes can be shown by their profiles.
```{r}
## --------------------------------------------------------------------------
betas <- coef(ddsTC)
colnames(betas)
```

## The log2 fold changes are plotted to a heatmap.
```{r}
## ----fissionheatmap--------------------------------------------------------
topGenes <- head(order(resTC$padj),20)
mat <- betas[topGenes, -c(1,2)]
thr <- 3 
mat[mat < -thr] <- -thr
mat[mat > thr] <- thr
pheatmap(mat, breaks=seq(from=-thr, to=thr, length=101),
         cluster_col=FALSE)
```

## The information about the analysis pipeline is shown.
```{r}
## --------------------------------------------------------------------------
sessionInfo()
```

